core/api/service/base_request.hpp

```diff
@@ -76,18 +82,41 @@ namespace kagome::api::chain::request {
     }
 
    private:
-    KAGOME_LOAD_VALUE(int32_t) {
-      if (!src.IsInteger32())
-        throw jsonrpc::InvalidParametersFault("invalid argument");
-
-      dst = src.AsInteger32();
+    template <typename T, typename = std::enable_if<std::is_integral_v<T>>>
+    void loadValue(boost::optional<T> &dst, const jsonrpc::Value &src) {
+      if (!src.IsNil()) {
+        T t;
+        loadValue(t, src);
+        dst = std::move(t);
+      } else {
+        dst = boost::none;
+      }
     }
 
-    KAGOME_LOAD_VALUE(int64_t) {
-      if (!src.IsInteger64())
-        throw jsonrpc::InvalidParametersFault("invalid argument");
+    template <typename T, typename = std::enable_if<std::is_integral_v<T>>>
+    void loadValue(T &dst, const jsonrpc::Value &src) {
+      if (!src.IsInteger32() and !src.IsInteger64())
+        throw jsonrpc::InvalidParametersFault("invalid argument type");
```

Comment:
It seems no reason for this refactoring. Previous variant was much more simple to understand. More over this method is still valid for int64 and int32 implementations. The others must be overloaded additionally. And fields values limitations now checks twice: in jsonrpc and here.

Reply:
It's can be usefull to check value range for narrow and unsigned integer numbers.